service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -534,6 +455,102 @@ public synchronized void init(HiveConf hiveConf) {
     // Extra time for releasing the resources if timeout sets to 0
     ShutdownHookManager.addGracefulShutDownHook(() -> graceful_stop(),  timeout == 0 ? 30 : timeout);
   }
+  
+  private void addHAContextAttributes(HttpServer.Builder builder, HiveConf hiveConf) {
```

Comment:
`static`?

Reply:
The method uses 2 HS2 member variables: isLeader and hs2HARegistry.
What is the benefit of making this method static?

Reply:
It's hard to spot instance variables during review.